src/index.js

```diff
@@ -1,96 +1,231 @@
+    } = await DRIVE.files.list({
+        q: `name='${currentFolder}' and '${parentFolderId}' in parents and trashed=false`,
         fields: 'files(id)',
         includeItemsFromAllDrives: true,
         supportsAllDrives: true,
     });
 
+    actions.debug(`files: ${JSON.stringify(files)}`);
+
     if (files.length > 1) {
         throw new Error('More than one entry match the child folder name');
     }
     if (files.length === 1) {
-        return files[0].id;
+        actions.debug(`${currentFolder} exists inside ${parentFolderId}`);
+        // Folder exists, check that folders children
+        return getUploadFolderId(files[0].id, remainingFolderPath);
     }
 
-    const childFolderMetadata = {
-        name: childFolder,
+    actions.debug(`${currentFolder} does not exist inside ${parentFolderId}`);
+
+    const currentFolderMetadata = {
+        name: currentFolder,
         mimeType: 'application/vnd.google-apps.folder',
         parents: [parentFolderId],
     };
-    const { data: { id: childFolderId } } = await drive.files.create({
-        resource: childFolderMetadata,
+    const {
+        data: { id: currentFolderId },
+    } = await DRIVE.files.create({
+        requestBody: currentFolderMetadata,
         fields: 'id',
         supportsAllDrives: true,
     });
 
-    return childFolderId;
-}
+    actions.debug(`${currentFolder} id: ${currentFolderId}`);
 
-async function main() {
-    const uploadFolderId = await getUploadFolderId();
+    return getUploadFolderId(currentFolderId, remainingFolderPath);
+}
 
-    if (!filename) {
-        filename = target.split('/').pop();
-    }
+/**
+ *  Uploads a file from the filesystem
+ *
+ * @param {string} fileName Name to use in Google Drive
+ * @param {string} filePath Path to the file on the filesystem
+ * @param {boolean} override Whether or not to remove and replace the current file if it exists
+ * @param {string} uploadFolderId Id of the new files parent
+ * @returns {Object} Response from the google drive files create api
+ */
+async function uploadFile(fileName, filePath, override, uploadFolderId) {
+    console.log(`Uploading ${fileName} ...`);
+    actions.debug(`fileName: ${fileName}`);
+    actions.debug(`filePath: ${filePath}`);
+    actions.debug(`override: ${override}`);
+    actions.debug(`uploadFolderId: ${uploadFolderId}`);
 
     const fileMetadata = {
-        name: filename,
+        name: fileName,
         parents: [uploadFolderId],
     };
+
     const fileData = {
-        body: fs.createReadStream(target),
+        body: fs.createReadStream(filePath),
     };
 
     if (override) {
-        const res = await drive.files.list({
+        const { files } = await DRIVE.files.list({
             q: `'${uploadFolderId}' in parents`,
             fields: 'nextPageToken, files(id, name)',
         });
 
-        const { files } = res.data;
-
-        files.forEach((file) => {
-            if (file.name === filename) {
+        for (const file in files) {
```

Comment:
```suggestion
        for (const file of files) {
```

Currently iterates over object keys rather than the array members.

Reply:
Ya kind of overlooked that.